### PR TITLE
Certificate renewal instruction updated using dns chalange approach.

### PR DESCRIPTION
**Pull request by sakalosj**
Certificate renewal instruction updated using dns chalange approach.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -272,92 +274,73 @@ pro-eu-west-1-infra-1781350677.eu-west-1.elb.amazonaws.com. 60 IN A 52.30.203.24
...
+Check if you have access to packit.dev domain in [Google Domains](https://domains.google.com/m/registrar/packit.dev).
 
-```
-$ oc describe route.route.openshift.io/get-them-certs
-Name:                   get-them-certs
-Namespace:              packit-prod
-Requested Host:         prod.packit.dev
-                          exposed on router router (host elb.e4ff.pro-eu-west-1.openshiftapps.com) 10 minutes ago
-```
+Cehck/install certbot localy [certbot](https://certbot.eff.org/all-instructions).
```
Comment (csomh) [inline README.md:279]:
```suggestion
Check/install [certbot](https://certbot.eff.org/all-instructions) locally.
```

File: README.md [hunk 2]
```diff
@@ -272,91 +274,80 @@ pro-eu-west-1-infra-1781350677.eu-west-1.elb.amazonaws.com. 60 IN A 52.30.203.24
...
-Verify the `get-them-certs` route exposes the `prod.packit.dev`:
+Run certbot:
 
 ```
-$ oc describe route.route.openshift.io/get-them-certs
-Name:                   get-them-certs
-Namespace:              packit-prod
-Requested Host:         prod.packit.dev
-                          exposed on router router (host elb.e4ff.pro-eu-west-1.openshiftapps.com) 10 minutes ago
+certbot certonly --config-dir /var/tmp --work-dir /var/tmp --logs-dir /var/tmp --manual --preferred-challenges dns --email user-cont-team@redhat.com -d *.packit.dev -d *.prod.packit.dev -d *.stg.packit.dev
```
Comment (csomh) [inline README.md:286]: Would `*.packit.dev` be enough?

File: README.md [hunk 3]
```diff
@@ -272,91 +274,80 @@ pro-eu-west-1-infra-1781350677.eu-west-1.elb.amazonaws.com. 60 IN A 52.30.203.24
...
+Press Enter to Continue
 ```
-$ export CERTS_POD=$(oc get pods | grep certs | awk '{print $1}')
-$ oc rsh ${CERTS_POD} certbot certonly --config-dir /tmp --work-dir /tmp --logs-dir /tmp --manual --email user-cont-team@redhat.com -d prod.packit.dev
 
-...
+### 3. Update DNS record
 
-Create a file containing just this data:
+Go to [Google Domains](https://domains.google.com/m/registrar/packit.dev/dns) and create/set correspondig value: TXT record type with name \_acme-challenge/\_acme-challenge.prod/\_acme-challenge.stg
```
Comment (csomh) [inline README.md:305]:
```suggestion
Go to [Google Domains](https://domains.google.com/m/registrar/packit.dev/dns) and create/set the corresponding value: TXT record called \_acme-challenge/\_acme-challenge.prod/\_acme-challenge.stg
```

File: README.md [hunk 4]
```diff
@@ -272,91 +274,80 @@ pro-eu-west-1-infra-1781350677.eu-west-1.elb.amazonaws.com. 60 IN A 52.30.203.24
...
+```
+nslookup -q=TXT _acme-challenge.prod.packit.dev
 ```
 
-and answer questions until it tells you to create a file containing data **abc** and make it available on your web server at URL http://prod.packit.dev/.well-known/acme-challenge/xyz.
-Don't close the session yet!!!
-
-Now we need to serve the challenge using the [haxxx/serve-acme-challenge.py](./haxxx/serve-acme-challenge.py) script.
-We'll use another session for that. First, copy the script into the pod:
+once it will return configured value
```
Comment (csomh) [inline README.md:313]:
```suggestion
once it returns the configured value
```

File: README.md [hunk 5]
```diff
@@ -272,91 +274,80 @@ pro-eu-west-1-infra-1781350677.eu-west-1.elb.amazonaws.com. 60 IN A 52.30.203.24
...
 
-```
-$ oc rsh ${CERTS_POD} python3 /tmp/haxxx/serve-acme-challenge.py /.well-known/acme-challenge/xyz abc
+Authoritative answers can be found from:
+
+[~/]$ ping dashboard.packit.dev
 ```
 
-Values of `<abc>` and `<xyz>` are generated by cert-bot in the first terminal session.
+Go to terminal wtih certbot command waiting for your action and hit Enter
```
Comment (csomh) [inline README.md:328]:
```suggestion
Go to the terminal with certbot command waiting for your action and hit Enter.
```

File: README.md [hunk 6]
```diff
@@ -272,91 +274,80 @@ pro-eu-west-1-infra-1781350677.eu-west-1.elb.amazonaws.com. 60 IN A 52.30.203.24
...
 
-Return to first terminal and Press Enter to Continue.
+Repeat this for all requested domains.
 
-And that's it! We got the certs now, let's get them from the pod to this git repo:
+Once finished change owner and copy certificates to secrets repository (prod&stg)
```
Comment (csomh) [inline README.md:332]: Hm, why is this needed?

File: README.md [hunk 7]
```diff
@@ -251,13 +251,15 @@ is a ton of places for improvements and ideally make it automated 100%.
...
-1. Deploy a certbot container in the project.
-2. Use the manual challenge.
-3. Run certbot and obtain the challenge.
-4. Run a custom script and serve the secret.
-5. Collect the certs.
+1. Check prerequisites.
+2. Run certbot to obtain the challenges.
+3. Configure DNS TXT records based on values requested in 2.
+4. Re-deploy stg&prod.
+5. Update secrets repository.
```
Comment (csomh) [inline README.md:258]: Could we switch these two in order to make sure, that the new cert is saved and available to the rest of the team?

File: README.md (hunk 2 above)
Comment (sakalosj) [reply README.md:286]: no, i have tried. Also you can check [so link](https://stackoverflow.com/questions/26744696/ssl-multilevel-subdomain-wildcard)

Comment (csomh) [reply README.md:286]: Gotcha! I wouldn't have expected this, though it makes sense... Would making a note about this here help avoiding confusion or it would be superfluous?

Comment (sakalosj) [reply README.md:286]: was thinking if it should be included, so will put it here

File: README.md (hunk 6 above)
Comment (sakalosj) [reply README.md:332]: certbot requires root permissions or change multiple write locations
```
[Errno 13] Permission denied: '/var/log/letsencrypt'
Either run as root, or set --config-dir, --work-dir, and --logs-dir to writeable paths.
```
i was thinking default, behavior is more secure and clean. WDYT?

File: README.md (hunk 7 above)
Comment (sakalosj) [reply README.md:258]: ok will switch them

File: README.md (hunk 6 above)
Comment (sakalosj) [reply README.md:332]: i will also incude point for executing certbot to run under if we will agree on this root approach

Comment (csomh) [reply README.md:332]: I think this is b/c the original use-case of certbot is to update certificates on live systems - in this case running as root makes sense. But if running in manual mode to get certs for *another* system, I would say there is no reason to run certbot as root.

I would rather say to change those paths to some user writable ones.

**Comment by sakalosj** [comment]
@TomasTomecek How exactly you want to perserve old approach? If we will keep it in "main" documentation it will make it long and maybe messy. Maybe link to older commit?

**Comment by TomasTomecek** [comment]
> @TomasTomecek How exactly you want to perserve old approach? If we will keep it in "main" documentation it will make it long and maybe messy. Maybe link to older commit?

That's a fair point. Link to the commit sounds good to me (or moving it to a separate file).

File: README.md (hunk 2 above)
Comment (sakalosj) [reply README.md]: updated

File: README.md (hunk 6 above)
Comment (sakalosj) [reply README.md]: updated to non root approach

File: README.md (hunk 7 above)
Comment (sakalosj) [reply README.md]: switched

**Comment by sakalosj** [comment]
> > @TomasTomecek How exactly you want to perserve old approach? If we will keep it in "main" documentation it will make it long and maybe messy. Maybe link to older commit?
> 
> That's a fair point. Link to the commit sounds good to me (or moving it to a separate file).

added link as there should be low probability that it will be required as this approach is simpler

File: README.md [hunk 8]
```diff
@@ -249,15 +249,25 @@ For more information please refer to [official docs](https://ansible.softwarefac
 Please bear in mind this is the easiest process I was able to figure out: there
 is a ton of places for improvements and ideally make it automated 100%.
 
+We are using certificates multidomain wildcard certificate for following domains:
```
Comment (TomasTomecek) [inline README.md:252]:
```suggestion
We are using multi-domain wildcard certificates for following domains:
```

File: README.md [hunk 9]
```diff
@@ -249,15 +249,25 @@ For more information please refer to [official docs](https://ansible.softwarefac
 Please bear in mind this is the easiest process I was able to figure out: there
 is a ton of places for improvements and ideally make it automated 100%.
 
+We are using multi-domain wildcard certificates for following domains:
+_.packit.dev
+_.prod.packit.dev
+\*.stg.packit.dev
```
Comment (csomh) [inline README.md:255]:
```suggestion
- *.packit.dev
- *.prod.packit.dev
- *.stg.packit.dev
```

File: README.md [hunk 10]
```diff
@@ -249,15 +249,25 @@ For more information please refer to [official docs](https://ansible.softwarefac
...
+
+In case procedure bellow will not work, previously used http challenge can be used instead [link](https://github.com/packit-service/deployment/blob/008f5eaad69a620c54784f1fc19c7c775af9ec7d/README.md#obtaining-a-lets-encrypt-cert-using-certbot).
+Be aware that http challenge approach is more complex, includes destructive actions and longer downtime
```
Comment (csomh) [inline README.md:258]:
```suggestion
Be aware that http challenge approach is more complex, includes destructive actions and longer downtime.
```

File: README.md [hunk 11]
```diff
@@ -249,15 +249,25 @@ For more information please refer to [official docs](https://ansible.softwarefac
...
-4. Run a custom script and serve the secret.
-5. Collect the certs.
+1. Check prerequisites.
+2. Run certbot to obtain the challenges.
+3. Configure DNS TXT records based on values requested in 2.
+4. Update secrets repository.
+5. Re-deploy stg&prod.
 
-### Prep
+_Note: If certbot was excuted against multiple domains you will repeat step 3. for each domain_
```
Comment (csomh) [inline README.md:268]:
```suggestion
_Note: If certbot was excuted against multiple domains you will repeat step 3. for each domain._
```

File: README.md [hunk 12]
```diff
@@ -272,90 +282,77 @@ pro-eu-west-1-infra-1781350677.eu-west-1.elb.amazonaws.com. 60 IN A 52.30.203.24
...
-Don't forget to do cleanup:
-
-```
-$ oc delete all -l name=get-them-certs
-$ oc delete all -l service=get-them-certs
-```
+- push new branch and merge.
 
-And deploy the `packit-service` (route & new certs):
+### 5.Re-deploy dev and stg environment:
```
Comment (csomh) [inline README.md:351]: Should this be stg and prod instead?

Comment (sakalosj) [inline README.md:351]:
```suggestion
### 5.Re-deploy stg and prod environment:
```
